homeassistant/components/whirlpool/climate.py

```diff
@@ -107,8 +107,8 @@ class AirConEntity(ClimateEntity):
 
     def __init__(self, hass, said, name, backend_selector: BackendSelector, auth: Auth):
         """Initialize the entity."""
-        self._aircon = Aircon(backend_selector, auth, said, self.async_write_ha_state)
-
+        self._aircon = Aircon(backend_selector, auth, said)
+        self._aircon.register_attr_callback(self.async_write_ha_state)
```

Comment:
Entity state update callbacks should normally be registered in entity method `async_added_to_hass`. Updating state before the entity has been added is an error.

Reply:
With the current API, no callbacks occur until after .connect() is called in async_added_to_hass.  Will create a separate PR as it's better practice as you mention.